File: Algoritmos_e_Logica_de_Programacao/Exercicios_logica_em_Java/Diagonal_negativos/src/Diagonal_negativos.java
import java.util.Scanner;
import java.util.Locale;

public class Diagonal_negativos {
    public static void main(String[] args) throws Exception {

        Locale.setDefault(Locale.US);
        Scanner sc = new Scanner(System.in);

        System.out.print("Qual a ordem da matriz? ");
        int n = sc.nextInt();

        int[][] vet = new int[n][n];

        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                System.out.print("Elemento [" + i + "," + i + "]: ");
                vet[i][j] = sc.nextInt();
            }
        }

        System.out.println("Diagonal principal: ");
        for (int i = 0; i < n; i++) {
            System.out.print(" " + vet[i][i]);
        }

        int somaNegativos = 0;
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                if (vet[i][j] < 0) {
                    somaNegativos += 1;
                }
            }
        }

        System.out.println();
        System.out.print("Quantidade de negativos = " + somaNegativos);

        sc.close();
    }
}

File: Algoritmos_e_Logica_de_Programacao/Exercícios_logica_em_Python/Diagonal_negativos/Diagonal_negativos.py
n: int

print("Qual ordem da matriz? ", end="")
n = int(input())

x: int = [[0 for x in range(0, n)] for x in range(0, n)]

for i in range(0, n):
    for j in range(0, n):
        x[i][j] = int(input(f"Elemento [{i},{j}]: "))

print("Diagonal principal: ")

for i in range(0, n):
    print(x[i][i], end=" ")


negativos = 0
for i in range(0, n):
    for j in range(0, n):
        if x[i][j] < 0:
            negativos += 1 

print()
print(f"Quantidade negativos = {negativos}")